### DISCUSSION TITLE
Exporting Variables?

### DISCUSSION THREAD

**Discussion by cgatian** [marcomontalbano/figma-export#171] [Q&A]
Is there a way to export Figma Variables? It looks like the CLI only supports exporting styles from a file.

**Comment by riboher** [comment]
Hi @marcomontalbano! Not sure if this is going to be supported in the near future. It would be great to be able to download Figma Variables using this package. Any feedback on this? Thank you so much!

**Comment by marcomontalbano** [comment]
Hi!
Thanks a lot for your interest in having variable export support in figma-export.
Unfortunately, Figma’s variable API is only available for paid plans, and at the moment, I don’t have an active paid subscription that would allow me to implement this feature.

I’ve reached out to Figma to see if they can offer any support for open-source maintainers like me. In the meantime, I really appreciate your understanding and your support for the project!

If anything changes, I’ll make sure to share updates.

**Reply by riboher** [reply]
Hi there @marcomontalbano!

I'd like to resume this discussion offering our help in implementing the feature in the project. The company I'm working on would really like to make use of the Figma variables and we ourselves have a token to try the results.

Would that be a feature that would make sense, provided the consumer has a token theirselves? Of course, knowing that this is a paid feature, it would be something that not everyone would enjoy, but is the only thing that occurred to me as I understand this is not a priority in your roadmap.

We're also considering other options, but we'd love to hear from you and see what your thoughts are about this.

Thanks a lot!

**Reply by marcomontalbano** [reply]
Hi @riboher,  

I got an answer from Figma. They're actively exploring ways to enable development on these APIs by community contributors, but they don’t yet have a defined timeframe for a solution. So, unfortunately, their answer is: no :cry:.  

I really appreciate your suggestion, which was actually my second plan: finding someone who can help with this.  

I’m not sure if it’s possible for you to add me to the organization with restricted permissions and access to just one file (for example, a simple file with variables). I don’t know exactly how permissions work in Figma for enterprise customers, but this would be the best option since I could directly work on the feature.  

Another option could be helping me by sharing API responses, so that at least I can start implementing the feature, and then you can test the library within a dedicated pull request.

What do you think?

I know this will be an enterprise-only feature, but I would be happy to implement it. 

In the meantime, I forked this [official Figma file](https://www.figma.com/design/Zbd0s4J3y2dvOnxLEr6ikq/Get-started-with-variables--Community-?node-id=1-3827), which I think could be a good starting point.

**Reply by riboher** [reply]
Hi Marco! Let me consult the options with the team and I'll get back to you. We have a couple of meetings scheduled for next week to discuss the work we need to do related to this, I hope I can give you a proper answer around October 1st. Talk to you then!
